All latest changes

semik997 opened the pull request.

semik997 committed:
add new tabs
started setting project

semik997 committed:
1

semik997 committed:
ended setting main storyboard

semik997 committed:
story

semik997 committed:
ended setting main storyboard

semik997 committed:
merge main in collection-View-main

semik997 committed:
Replaced the table view on the main and selected screen with the collection view

semik997 committed:
Added a placeholder when there are no search results
Added Like button and movie title
Recovered cells

semik997 committed:
Started working on multithreading the application
Fixed bugs with placeholder for main
Added placeholder for favorite

semik997 committed:
added button ...

semik997 committed:
added button ... in favorite view, configured the button for two view

semik997 committed:
Added controller for viewing settings
Added settings cells
Added logic for changing the background color

semik997 committed:
Added colorComplition, try tune change background color main VC

semik997 committed:
Added delegate to set cell color and position
Added setting to change cell size
Created a settings menu on the storyboard

semik997 committed:
Added sharing option via standard sharing interface
Fixed bug with open details from main view

semik997 committed:
Fixed bug with change color background in info view

semik997 committed:
Fixed bug with change color background in main view

semik997 committed:
Added the ability to select the size and position of the cell on the screen
Updated the image in the placeholder

semik997 committed:
Stored cell position and background color after leaving setup mode
Code refactoring

semik997 committed:
Added rating parsing from https://rapidapi.com/apidojo/api/imdb8/ and display in a cell

semik997 committed:
changet link

semik997 committed:
change rating from favorite

semik997 committed:
Added the ability to share links to movies on the site https://www.imdb.com/

semik997 committed:
Started connecting Core Data and storing data in it
Added an improvement (when sharing, the link of the movie opens on IMDb, if it is there)
Fixed display of cells

semik997 committed:
I continue setting up CoreData
Prepared the base
Prepared reading from the database
I'm getting an error while saving

semik997 committed:
Made saving and reading in CoreData

semik997 committed:
Fixed bug with saved idFilm
Fixed a bug with opening the additional information window

semik997 committed:
Fixed removing a movie from favorites on the main screen
Fixed the fill button like if you liked the movie
Finished injecting CoreData into the app
Fixed deleting movies from favorites

semik997 committed:
Added offline mod

semik997 committed:
refactoring code

semik997 committed:
Refactored cod

semik997 committed:
Added Firebase

PaszhaBiceps commented on CoreData/FavoriteFilm+CoreDataProperties.swift:
```diff
@@ -0,0 +1,32 @@
+    @NSManaged public var name: String?
+    @NSManaged public var original: Data?
+    @NSManaged public var summary: String?
+    @NSManaged public var url: String?
```
Почему все поля вынесены в экстеншн и отельным файлом?
На будущее - кол-во экстеншнов лучше сокращать по возможности. Если их сильно много будет это отрицательно скажется на времени компиляции.

semik997 replied:
Это создалось автоматически, я в ручную изменения не вносил

PaszhaBiceps replied:
Тогда перенеси все проперти в сам класс и удали лишние экстеншны, они тут ни к чему.

PaszhaBiceps commented on CoreData/FavoriteFilm+CoreDataProperties.swift:
```diff
@@ -0,0 +1,32 @@
+
+}
+
+extension FavoriteFilm : Identifiable {
```
Наследование протоколов лучше делать для основного класса.
И тут вопрос - зачем этому объекту быть Identifiable?

semik997 replied:
Файл был создан автоматически, ничего сам не вносил

PaszhaBiceps commented on Movie Search App.xcodeproj/xcuserdata/semenkolesnikov.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:
```diff
@@ -180,5 +180,19 @@
             landmarkType = "7">
          </BreakpointContent>
       </BreakpointProxy>
+      <BreakpointProxy
```
Займись файлом gitingnore, что бы этого всего не было в комметах.

PaszhaBiceps commented on Movie Search App.xcodeproj/xcuserdata/semenkolesnikov.xcuserdatad/xcschemes/xcschememanagement.plist:
```diff
@@ -14,6 +14,27 @@
 			<key>orderHint</key>
 			<integer>0</integer>
 		</dict>
+		<key>Promises (Playground) 1.xcscheme</key>
```
gitingnore

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+    }
+    
+    private func getContext() -> NSManagedObjectContext {
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
```
Не ассерти каст shared.delegate в AppDelegate. Это потенциальный креш.

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+    
+    private func getContext() -> NSManagedObjectContext {
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
+        return appDelegate.persistentContainer.viewContext
```
Зачем в ячейке контекст кор даты? Его тут быть не должно?

semik997 replied:
Убрал

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+        return appDelegate.persistentContainer.viewContext
+    }
+    
+    @IBAction func likeButton(_ sender: UIButton) {
```
Старайся делать функции, аутлеты и экшны - приватными (если тебе не нужно будет использовать что либо вне ячейки).

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+            //delited like
+            fillButton.isSelected = false
+            isFavorite = !isFavorite
+            delegate?.selectCell(isFavorite, idFilm: idFilm, url: url, name: name,
```
Зачем в делегате возвращать ВСЕ что есть если по факту нужны айди и новый статус?

semik997 replied:
Что бы не создавать новый делегат с меньшими 2мя переменными, которые передаются в том который есть

PaszhaBiceps replied:
А зачем изначально нужна вся инфа? По хорошему если тебе нужен конкретный объект - передай айдишник нужного фильма/обьекта и достань его из массива в делегате.
И если ты используешь одну и ту же функцию делегата для двух разных действий - лучше так не делать (при работе в команде или если после тебя кто-то будет что-то делать это может вводить в заблуждение и приведет к трате времени на ненужный инвестигейт и курение кода). Лучше расширять делегат по мере необходимости и передать в функции минимально необходимое кол-во инфы/параметров.

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+        if isFavorite {
+            //delited like
+            fillButton.isSelected = false
+            isFavorite = !isFavorite
```
Это должно храниться в модели (да, я помню что там оно уже есть) и модифицироваться вне ячейки.

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+        summary = film.show?.summary
+        imdb = film.show?.externals?.imdb
+        fillButton.isSelected = false
+        self.secondAPI.fetchShowRaiting(forShow: imdb ?? "") { [self] currentShowIMDb in
```
Апишка не должна вызываться из ячейки, по крайней мере этот случай так точно. Сколько запросов у тебя уйдет если в выдаче будет 1000 ячеек?
Эту информацию ты должен получать вне ячейки и просто ее туда отдавать.

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+        ratingLabel.text = "\(rating ?? 0)/10"
+    }
+    
+    func loadDataFavorite(film: FavoriteFilm) {
```
Нужно сделать что бы у тебя была одна моделька для всего. Потому что по факту что обычная моделька, что фаворит - это одно и тоже.

semik997 replied:
Разные, так как на меин экран загружаются данные с парсинга, а с фаворт загружаются данные из кор даты

PaszhaBiceps replied:
Ты же можешь избежать этого сделай init в одной модельку с помощью другой (для обоих классов). Тем самым ты будешь работать, скажем, только с Film, а FavoriteFilm нужна будет толко что бы достать и сохранить данные в кор дату.
Обдумай это и сделай красиво 😏

PaszhaBiceps commented on Movie Search App/CollectionVC/CollectionViewCell.swift:
```diff
@@ -0,0 +1,137 @@
+    
+    // MARK: - String in image conversion
+    
+    func getImage(from string: String) -> UIImage? {
```
Возвращай здесь в случае ошибок и прочего плейсхолдер вместо реальной картинки.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+    }
+    
+    private func getContext() -> NSManagedObjectContext {
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
```
Снова форс анврап аппарат делегата.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        return text.isEmpty
+    }
+    private var filtredFilms: [FavoriteFilm] = []
+    var filmsFav: [FavoriteFilm] = []
```
Как я писал выше - делай все проперти приватными.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+                fatalError("Unresolved error \(nserror), \(nserror.userInfo)")
+            }
+            self.filmsFav.remove(at: index)
+            self.favoriteCollectionView.reloadData()
```
Вынеси всю работу из экшна в отдельную функция и вызывай именно ее. Так будет чище.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        }
+        
+        let no = UIAlertAction(title: "No thanks", style: .cancel){ action in
+            self.favoriteCollectionView.reloadData()
```
Зачем перезагружать таблицу в этом случае?

semik997 replied:
Что бы вернуть заполненную кнопку лайка а не оставлять пустой

PaszhaBiceps replied:
То есть ты по тапу на кнопку сразу меняешь состояние? Учитывая что у тебя есть алерт что бы подтвердить действие это можно не делать и тем самым лишний раз ничего не перезагружать.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        let no = UIAlertAction(title: "No thanks", style: .cancel){ action in
+            self.favoriteCollectionView.reloadData()
+        }
+        alertController.addAction(yes)
```
Если хочешь прямо красиво - можно не инициализировать экшны отдельно, а создавать их прямо здесь.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+            guard let indexPath = favoriteCollectionView.indexPathsForSelectedItems else { return }
+            let film: FavoriteFilm
+            if isFiltering {
+                film = filtredFilms[indexPath[0].row]
```
Почему здесь всегда берется 0 индекс? А что будет если массив filtredFilms пустой?

semik997 replied:
То соответственно некуда будет переходить и нечего открывать, так как этот сигвей открывает подробную информацию о фильме
Если элементов не будет, значит и нечего будет открывать и некуда переходить

PaszhaBiceps replied:
В общем подумай как сделать переходы чище. Потому что сейчас ты зависишь от выбранных ячеек, а не должен.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+            if isFiltering {
+                film = filtredFilms[indexPath[0].row]
+            } else {
+                film = filmsFav[indexPath[0].row]
```
Аналогично с верхним.

semik997 replied:
Так же, если нет фильмов, то и искать нечего и ничего не отобразиться

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+                film = filmsFav[indexPath[0].row]
+            }
+            let nav = segue.destination as! UINavigationController
+            let MoreInfoFavoritesTableVC = nav.topViewController as! MoreInfoViewController
```
Два форс анврапа подряд, поправь.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        
+        //MARK: - Info button
+        
+        if segue.identifier == "popVC" {
```
Вынеси когда будет время идентификаторы сегью в константы, так будет чище и лучше.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        }
+        self.big = big
+        self.medium = medium
+        self.small = small
```
Сделай enum который будет отвечать за состояние интерфейса. 3 бульки выглядят страшно.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+        
+        if isFavorite == false {
+            //for not like
+            presentAlertController(withTitle: "Do you sure?", message: nil, style: .alert,
```
Неправильно написал тайтл.

semik997 replied:
Не совсем понял замечание, можешь, пожалуйста, подробнее объяснить что не верно написано

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,259 @@
+            return defaultSizeCell
+        } else {
+            // apply small size cell
+            if self.small ?? false {
```
Возвращаемся к enum для размера. С ним это будет выглядеть намного чище.

PaszhaBiceps commented on Movie Search App/CollectionVC/InfoTableViewController.swift:
```diff
@@ -0,0 +1,66 @@
+    }
+    
+    override func viewWillLayoutSubviews() {
+        preferredContentSize = CGSize(width: 170, height: tableView.contentSize.height)
```
Почему тут не вызывается super.viewWillLayoutSubviews?

PaszhaBiceps commented on Movie Search App/CollectionVC/InfoTableViewController.swift:
```diff
@@ -0,0 +1,66 @@
+    
+    // setup size view
+    override var preferredContentSize : CGSize
+    {
```
Не переноси скобки на другую строчку без необходимости, выглядит так себе.

PaszhaBiceps commented on Movie Search App/CollectionVC/InfoTableViewController.swift:
```diff
@@ -0,0 +1,66 @@
+    //MARK: - Setting view
+    
+    // exit
+    @IBAction func exit(_ sender: UIButton) {
```
Наведи порядок в коде. У тебя в перемешку и делегаты и все что хочешь. И в целом приведи структуру экранов и прочих сущностей к одному знаменателю.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    
+    //Creating and filling the array for Display
+    var networkManager = NetworkManager()
+    var secondAPI = SecondAPI()
```
1. Сделай все приватным, как я уже несколько раз писал выше.
2. Что таке networkManager понятно. А что такое SecondAPI? За что она отвечает?

semik997 replied:
SecondAPI отвечает за рейтинг imdb, который я дополнительно запрашиваю с RapidAPI

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    var searchText = ""
+    var small: Bool?
+    var medium: Bool?
+    var big: Bool?
```
Enum.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    func getDataFromParse() {
+        
+        do {
+            self.oneFilm = try context.fetch(FavoriteFilm.fetchRequest())
```
Название проперти не отражает того что она делает. И сразу вопрос - зачем это тут?

semik997 replied:
Лишняя, убрал

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    }
+    
+    private func getContext() -> NSManagedObjectContext {
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
```
Снова форс анврап. И вообще, если тебе нужно взаимодействовать с кор датой в нескольких местах, не проще ли сделать отдельную сущность которая будет отвечать за это и предоставить другим классам только некоторые вещи?

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    override func collectionView(_ collectionView: UICollectionView,
+                                 numberOfItemsInSection section: Int) -> Int {
+        if films.count == 0 {
+            findImage.isHidden = false
```
Опять таки неймниг, за что отвечает findImage?

semik997 replied:
Отображение картинки бек граунда, если нет никакой информации то она видима, если есть хотя бы 1 результат, то невидима

PaszhaBiceps replied:
Назови ее тогда чем то вроде noContentImageView, что бы было понятно за что она отвечает. И вообще (как я уже писал) - следи за неймингом. Имя переменной должно отражать то за что она отвечает.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    func  presentInternetConnectionAlertController () {
+        let internetAlert = UIAlertController(title: "No Internet Connection", message: "Make sure your device is connected to the internet.", preferredStyle: .alert)
+        let ok = UIAlertAction(title: "OK", style: .cancel){ action in
+            self.collectionViewSpace.reloadData()
```
Зачем здесь перезагрузка коллекции?

semik997 replied:
Что-то на подобии видимой перезагрузки экрана для пользователя
Убрать?

PaszhaBiceps replied:
Можешь оставить если считаешь что это реально нужно.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "showDetail" {
+            guard let indexPath = collectionViewSpace.indexPathsForSelectedItems else { return }
+            let film = films[indexPath[0].row]
```
Почему здесь берется именно 0 индекс?

semik997 replied:
Индексация экрана с 0 элемента
Так же без 0 индекса пишет что row не принадлежит indexPath

PaszhaBiceps replied:
Логично, так как ты из коллекции получаешь массив IndexPath.  Сделай тогда проверку пустой массив или нет.
И вопрос - у твоей коллекции сингл или мультипл селекшн?

semik997 replied:
Проверку сделаю
Я правильно ли понимаю твой вопрос:
"И вопрос - у твоей коллекции сингл или мультипл селекшн?"
 Ты имеешь в виду можно ли у меня выбрать несколько ячеек или только 1?

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+            if searchText != "" {
+                let text = searchText.split(separator: " ").joined(separator: "%20")
+                findImage.isHidden = true
+                // Check count symbol
```
Старайся меньше комментировать очевидный код.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+                findImage.isHidden = true
+                // Check count symbol
+                if searchText.count >= 3 {
+                    // need add timer in this fragment
```
В таких случаях оставляй себе TODO. Что бы они прям тебе о себе напоминали, пиши их так - #warning("TODO: *Some stuff to do*").

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+            likeFilms.original = originalImageData
+            likeFilms.medium = imageData
+            likeFilms.summary = summary
+            likeFilms.imdb = imdb
```
Зачем здесь перезаписывать все поля?

semik997 replied:
Происходит перезапись для сохранения в кор дату

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+        } catch {
+            print(error.localizedDescription)
+        }
+        return image
```
Тут тоже - плейсхолдер в случае ошибки и т.п.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+            return defaultSizeCell
+        } else {
+            // apply small size cell
+            if self.small ?? false {
```
Enum.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+
+extension MainCollectionVC {
+    
+    func NKPlaceholderImage(image: UIImage?, imageView: UIImageView?, imgUrl: String, compate:@escaping (UIImage?) -> Void) {
```
Что это и зачем?

semik997 replied:
Функция для скачивания картинок и последующего сохранения их в кор дату, что бы можно было просматривать их в оффлайн режиме

PaszhaBiceps commented on Movie Search App/CollectionVC/SettingViewController.swift:
```diff
@@ -0,0 +1,72 @@
+    @IBOutlet weak var smallSizeCellButton: UIButton!
+    
+    weak var delegate: SettingViewControllerDelegate?
+    var small = false
```
Enum.

PaszhaBiceps commented on Movie Search App/Delegate/Film.swift:
```diff
@@ -0,0 +1,104 @@
+        let image: Image?
+        let externals: Externals?
+        let summary: String?
+        var isFavorite: Bool?
```
Вынеси это в Film через кастомные декодер, как я тебе и говорил.

PaszhaBiceps commented on Movie Search App/Delegate/Film.swift:
```diff
@@ -0,0 +1,104 @@
+        
+        get {
+            if let data = defaults.value(forKey: "favoriteFilm") {
+                return try! PropertyListDecoder().decode([Films.Film].self, from: data as! Data)
```
Не ассерти. Как я уже писал - это потенциальные креши.

PaszhaBiceps commented on Movie Search App/Delegate/Film.swift:
```diff
@@ -0,0 +1,104 @@
+//        } catch let error as NSError {
+//            print(error.localizedDescription)
+//        }
+//    }
```
Чисти комменты и закомментированный код если они не нужны и не несут никакой ценности.

PaszhaBiceps commented on Movie Search App/Delegate/SceneDelegate.swift:
```diff
@@ -13,10 +13,10 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
-        // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
         guard let _ = (scene as? UIWindowScene) else { return }
+        let mvc = window?.rootViewController as? MainCollectionVC
+        let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
```
Снова ассершн.

PaszhaBiceps commented on Movie Search App/Delegate/SecondAPI.swift:
```diff
@@ -0,0 +1,53 @@
+class SecondAPI {
+    
+    // MARK: - Fetch show raiting
+    func fetchShowRaiting(forShow show: String, completionHandler: @escaping (Films.FilmIMDb) -> Void) {
```
Зачем отдельный аги класс для одного запроса (и парсер)? Перенеси это все в NetworkingManager.

semik997 replied:
Так как парсит с другого апи и по другой структуре

PaszhaBiceps commented on Movie Search App/MoreInfoViewController.swift:
```diff
@@ -9,50 +9,110 @@ import UIKit
+                application.open(appURL! as URL)
+            } else {
+                // if Youtube app is not installed, open URL inside Safari
+                application.open(webURL! as URL)
```
Снова ассершны.

PaszhaBiceps commented on Movie Search App/MoreInfoViewController.swift:
```diff
@@ -9,50 +9,110 @@ import UIKit
+                application.open(appURL! as URL)
+            } else {
+                // if Youtube app is not installed, open URL inside Safari
+                application.open(webURL! as URL)
```
Ассрешны.

PaszhaBiceps reviewed (changes requested):
Пройдись по комментам и пофиксил. И глобально - избегай ассершнов, приведи экраны к одно структуре (что бы каши не было) и старайся делать меньше экстернов (особенно там где это не нужно). Экстешны чудесно меняются на марки и ты не теряешь в читаемости.

semik997 committed:
Added new gitignore, refactor code

semik997 committed:
Made, where necessary, functions and variables private

semik997 committed:
add enum, refactoring code

semik997 committed:
Removed!
Added a placeholder instead of an error if there is no image
Made an enum for cell sizes
Added constants

semik997 committed:
rename API files

semik997 committed:
fixed bug with constraints and flood in console

semik997 committed:
Changed the file structure, add pod file

PaszhaBiceps commented on Movie Search App/API/RapidApiManager.swift:
```diff
@@ -0,0 +1,53 @@
+    }
+    
+    // Call completion block with json
+    private func parseJSONRaitinig(withData data: Data) -> Films.FilmIMDb? {
```
Как сделаешь что что делаешь сейчас - напиши универсальный парсер для Codable объектов (подсказка - на дженериках). Что бы избежать подобного дублирования кода в дальнейшем.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,274 @@
+        }
+    }
+    
+    private func getContext() -> NSManagedObjectContext {
```
Я там где-то писал по поводу сущности которая будет взаимодействовать с кор датой. Ты не делал это еще?
Ну и по классике - ассершн убери 👌

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -0,0 +1,349 @@
+    }
+
+    
+//    func uploadImage(photo: UIImage, completion: @escaping (Result<URL, Error>) -> Void) {
```
Если это не нужно - удали.

PaszhaBiceps commented on Movie Search App/Model+Constants/Constants.swift:
```diff
@@ -1,13 +1,24 @@
+
+    // MARK: - Constants
+
+let showDetail = "showDetail"
```
Закатай это в структуру типа Segues, что бы было понятнее что это и зачем.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -0,0 +1,274 @@
+
+extension FavoritesCollectionVC: UICollectionViewDelegateFlowLayout {
+    
+    enum ChooseSize {
```
Да, ты в итоге сделал не так enum как я имел ввиду. Смотри, ты должен передавать не 3 бульки, а 1 енам кейс. И по этому кейсу ты уже будешь ставить нужный сайз (сам размер сделай комьпьютед велью самого енама).

PaszhaBiceps reviewed (changes requested):
Пробежался - уже лучше. Я добавил еще пачку комментов - пройдись по ним.

semik997 committed:
Added new delegate for delete film on favorite
Refactoring code

semik997 committed:
fix bug with delet film

semik997 committed:
Fixed enum from choose size from cell, added struct for constans
